tests/test_CompareEclipseRestart.cpp

```diff
@@ -0,0 +1,173 @@
+#include "test_CompareEclipseRestart.hpp"
+
+int main(int argc, char** argv) {
+    char* gridFile1 = argv[1];
+    char* gridFile2 = argv[2];
+    char* unrstFile1 = argv[3];
+    char* unrstFile2 = argv[4];
+
+    if (argc != 5) {
+        printHelp();
+    }
+    else {
+        // Comparing grid sizes from .EGRID-files
+        {
+            ecl_grid_type* ecl_grid1 = ecl_grid_alloc(gridFile1);
```

Comment:
> since [`ecl_grid_free(ecl_grid)`] changes `ecl_grid` I presume (?)

Only if it is poorly implemented.  The typical implementation of a function that disposes of a single dynamically allocated structure is something along the lines of
```C
void dispose(struct S *s)
{
    if (s != NULL) {
        for (all allocated fields f) {
            dispose_field(f);
        }
    }

    free(s);
}
```
If it were not for the fact that the C standard mistakenly declares `free()` as
```C
void free(void *ptr);
```
as opposed to `free(const void *ptr)` the above `dispose()` function really should be declared as
```C
void dispose(const struct S *s);
```